### PR TITLE
Automation task: ReqnRoll BDD framework with POM and feature files

### PR DESCRIPTION
**Pull request by testwithAagy**
Added automation for Home Page, Login Page  and  Languages (add, update, delete) & Skills(add, update, delete) sections in the profile page as per the user story.
Included positive testing with valid inputs, negative testing with valid inputs, negative testing with invalid inputs , and negative  destructive/malicious test scenarios.
Implemented scenario outlines with tables to pass values from feature files to steps.
Added hooks to reset state before tests (delete existing data) and clean up after each test (remove only data created by that test).Ensures tests are independent and can run in parallel without false positives/negatives.
All scenarios now add the required data before performing actions (update/delete) instead of assuming prior state.
Removed obsolete files and Profile feature.
Improved code structure and readability in steps, pages, and hooks.
Updated README to reflect all new features, hooks, and test scenarios.
Feature files now include Scenario Outlines with examples for parameterized testing as per your feedback
Extended test coverage -all scenarios include parameterized tests using Scenario Outline and Examples tables to ensure comprehensive coverage of positive, negative, and destructive cases.
All known bugs identified during testing are listed in the readme.md file under the KNOWN BUGS/ISSUES section, a separate detailed bug report is not created as you suggested.

### CODE REVIEW

File: qa-dotnet-cucumber/Features/Login.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Feature: Login Functionality
+  As a user, I want to log in to the application to access restricted content.
+
```
Comment (amanbansalnz) [inline qa-dotnet-cucumber/Features/Login.feature:4]: Hey can you please submit git url repo or pr link rather then zip file for future reference! Also your test cases are missing Please increase your coverage around the features you are testing here is an idea of what is missing 



1 

Please increase your coverage around the features you  are testing here is an idea of what is missing 

• Negative testing with valid input
• Negative testing with invalid input
• Destructive testing

E.g of above scenarios
Happy path tests check basic functionality and the acceptance criteria of the API. We later extend positive tests to include optional parameters and extra functionality.


The next group of tests is negative testing where we expect the application to gracefully handle problem scenarios with both valid user input (for example, trying to add an existing username) and invalid user input (trying to add a username which is null).

Think about remove something that does not exist, updating something that does not exist.

Destructive testing is a deeper form of negative testing where we intentionally attempt to break the API to check its robustness (for example, sending a huge payload body in an attempt to overflow the system).

1.1/
Please use tables  and pass values to your scenarios for example look at the below scenario 
This way you will get exposure to use them  and will know how to pass test data to the tests
You all need to pass the values down from feature files to the steps see examples below


Example 

	Given I logged into the portal
	When I click on tab Education
	And I add my education 
	Then I am able to see my education'

Would  change to 


@Education
Scenario Outline: 1-Add education
	Given I logged into the portal
	When I click on tab Education
	And I add my education including '<Country>', '<University>', '<Title>', '<Degree>', '<Graduation Year>'
	Then I am able to see my education details including '<Country>', '<University>', '<Title>', '<Degree>', '<Graduation Year>'

	Examples:
	| Country     | University                      | Title  | Degree  | Graduation Year |
	| New Zealand | Whitireia Community Polytechnic | M.Tech | Diploma | 2021            |
	| Venezuela   | Hhutech University              | B.A    | Diploma | 2011            |

File: qa-dotnet-cucumber/Features/Login.feature [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  Scenario: Perform a successful login
+    Given I am on the login page
+    When I enter valid credentials
+    Then I should see the secure area
+
+  Scenario: Fail login with invalid username
```
Comment (amanbansalnz) [inline qa-dotnet-cucumber/Features/Login.feature]: please add 



2 State and Clean up 


State

State is when one test does not depend on another for example test 1 adds language  English and test 2 updates the language English to French, then test 2 should not assume that  English is added by test 1 or any other language! Instead test 2 should first add the language it wants to update and then update it.  (Test 2 adds English and then updates it). This ensures that the test is in the correct state for the test to execute.

Cleanup
example before adding language you need to make sure there are no existing languages as this could result in false positive test pass you will need to think about this for all the scenarios, this should be done before you start any tests
the state is say that you have a test that is adding a language english, the second time you run this test you should delete any existing language so that the state is reset, this will guarantee that your test does not produce false negatives and passes always. You can not assume that test execution order is not always the same, imagine if you run the tests  in parallel mode the all tests will run at the same time so you need to also make sure you do not wipe out other tests data hence why when you remove data that other tests are using. The best thing to do is delete only data you added for the test to be correct e.g test 1 added english and french as language test 2 added hindi then when you reset data you already know what language you are adding for each test there for on reset of the two test you should delete english and french for test 1 and hindi for test 2
So to sum up you should delete all data before the start of any test and then after each test you should delete the data that a particular test is adding! This is ensure that upon staring the test pack you have reset state and upon finishing each test deletes their data that they added! Think about first clean up before any test runs (you delete all data around this feature e.g before any language test run you delete all languages). Then after each test delete only data that the test has added! (test1 added english and test2 to added french then test1 should only delete english and test2 should delete french only)
also say you are updating a language then you must added language then update it do not assume the language is there! This applies for delete or any scenario
note you need to clean call data before any test runs and then clean data that the test generates at the end of each test! (do not clean all the data but only the one that the test is generating).
I would also suggest you read up on Nunit hooks because there will be a hook the you will need to make use of (which will clean all the test data for both cert and education before any test starts) and then your another hook that will execute at the end of each test which will ( only remove the test data that the test is adding)

File: qa-dotnet-cucumber/Features/Login.feature (hunk 1 above)
Comment (amanbansalnz) [reply qa-dotnet-cucumber/Features/Login.feature:4]: See https://cucumber.io/docs/gherkin/reference/
